src/services/userService.js

```diff
@@ -0,0 +1,36 @@
+const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
+
+const { validateEmail } = require("../util/validators");
+const userDao = require("../models/userDao");
+
+const signup = async (name, phone, email, password) => {
+  validateEmail(email);
+  if (!name || !phone || !password || !email) {
+    throw new Error("KEY_ERROR");
+  }
+
+  const user = await userDao.getUserByEmail(email);
+  if (user) {
+    throw new Error("DUPLICATED_EMAIL", 400);
+  }
+  const hashedPassword = await bcrypt.hash(password, 10);
+  return await userDao.createUser(name, phone, email, hashedPassword);
+};
+
+const signin = async (email, password) => {
+  validateEmail(email);
+  const user = await userDao.getUserByEmail(email);
+  const match = await bcrypt.compare(password, user.password);
+  if (!match) {
+    throw new Error("WRONG_PASSWORD", 401);
+  }
```

Comment:
- boolean을 판단하는 변수는 is_를 많이 사용합니다.
- 비밀번호가 틀렸다는 것을 직접적으로 알려주기 보다는 아래와 같이 표현해 주면 좋을 것 같습니다.

```suggestion
  const is_matched = await bcrypt.compare(password, user.password);
  
  if (!is_matched) {
    throw new Error("INVALID_USER", 401);
  }
```